.github/actions/publish-to-s3/action.yml

```diff
@@ -0,0 +1,97 @@
+name: Publish to S3
+description: Publishes the specified directory using aws s3 sync
+
+inputs:
+  AWS_REGION:
+    description: The region of your AWS bucket
+    default: ca-central-1
+    required: false
+  AWS_ROLE:
+    description: The AWS role to assume. This role must have write permission for your bucket
+    required: true
+  BUCKET_PATH:
+    description: The full path of your bucket, including subdirectories, to which to publish
+    required: true
+  BUILD_DIRECTORY:
```

Comment:
Nit: maybe `PUBLISH_DIRECTORY` instead of "build", which is more specific to this use case?